sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContext.cs

```diff
@@ -113,14 +113,33 @@ public partial interface IDynamoDBContext : IDisposable
         #endregion
 
         #region BatchGet
+
+        /// <summary>
+        /// Creates a strongly-typed BatchGet object, allowing
+        /// a batch-get operation against DynamoDB.
+        /// </summary>
+        /// <typeparam name="T">Type of objects to get</typeparam>
+        /// <returns>Empty strongly-typed BatchGet object</returns>
+        BatchGet<T> CreateBatchGet<T>();
+
         /// <summary>
         /// Creates a strongly-typed BatchGet object, allowing
         /// a batch-get operation against DynamoDB.
         /// </summary>
         /// <typeparam name="T">Type of objects to get</typeparam>
         /// <param name="operationConfig">Config object which can be used to override that table used.</param>
         /// <returns>Empty strongly-typed BatchGet object</returns>
-        BatchGet<T> CreateBatchGet<T>(DynamoDBOperationConfig operationConfig = null);
+        [Obsolete("Use the CreateBatchGet overload that takes BatchGetConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to BatchGet.")]
```

Comment:
why did you change the signature of the method?

Reply:
If we added something similar to what was there, with the new config as optional:
```
BatchGet<T> CreateBatchGet<T>(DynamoDBOperationConfig operationConfig = null);
BatchGet<T> CreateBatchGet<T>(BatchGetConfig batchGetConfig = null);
```

Then these two calls are both ambiguous:
```
context.CreateBatchGet<Foo>();
context.CreateBatchGet<Foo>(null);
```

If we didn't make the new config optional:
```
BatchGet<T> CreateBatchGet<T>(DynamoDBOperationConfig operationConfig = null);
BatchGet<T> CreateBatchGet<T>(BatchGetConfig batchGetConfig);
```

Then
```
context.CreateBatchGet<Foo>();     // Calling the now obsolete overload, which we don't want
context.CreateBatchGet<Foo>(null); // Still ambiguous
```

So for that `context.CreateBatchGet<Foo>();` case, I added a non-obsolete parameterless `CreateBatchGet<T>()`
```
BatchGet<T> CreateBatchGet<T>();
BatchGet<T> CreateBatchGet<T>(DynamoDBOperationConfig operationConfig = null);
BatchGet<T> CreateBatchGet<T>(BatchGetConfig batchGetConfig);
```

With that I was worried `context.CreateBatchGet<Foo>()` would still be ambiguous between the first two, but reviewing https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#12643-better-function-member again the parameterless one will win. So I'll add back the middle call. 

`context.CreateBatchGet<Foo>(null);` is still ambiguous, but I don't think we can do anything about that (nor should we, since that call doesn't add anything over `context.CreateBatchGet<Foo>()` that I can see